modules/gcp/external-dns/main.tf

```diff
@@ -0,0 +1,103 @@
+resource "google_service_account" "external_dns" {
+  count        = var.external_dns_enabled == true ? 1 : 0
+  project      = var.project_id
+  account_id   = "${var.name_prefix}-external-dns"
+}
+
+resource "google_dns_managed_zone" "cluster" {
+  count      = var.external_dns_enabled == true ? 1 : 0
+  project    = var.project_id
+  name       = "${var.cluster_name}-${local.zone_name}"
+  dns_name   = "${var.cluster_name}.${local.dns_name}."
+}
```

Comment:
How will this work? Are users supposed to create their DNS names with the cluster name prefix?

Reply:
at this stage, I have tested public zones as `gcp.sandbox.tetrate.io` and `azure.sandbox.tetrate.io`, i am ding `aws.sandbox.tetrate.io`, i have put the placeholders to support private zones, I think we should sort out private zones  management before merging this PR.